case/src/main/kotlin/com/ritense/case_/rest/dto/CaseWidgetTabDto.kt

```diff
@@ -29,11 +29,16 @@ data class CaseWidgetTabDto(
 ) {
     companion object {
         @JvmStatic
-        fun of(tab: CaseWidgetTab, widgetMappers: List<CaseWidgetMapper<CaseWidgetTabWidget, CaseWidgetTabWidgetDto>>): CaseWidgetTabDto {
+        fun of(
+            tab: CaseWidgetTab,
+            widgetMappers: List<CaseWidgetMapper<CaseWidgetTabWidget, CaseWidgetTabWidgetDto>>,
+            permissionCheck: (CaseWidgetTabWidget) -> Boolean
+        ): CaseWidgetTabDto {
             return CaseWidgetTabDto(
                 tab.id.caseDefinitionName,
                 tab.id.key,
                 widgets = tab.widgets
+                    .filter { permissionCheck(it) }
```

Comment:
Alternatively, we could do the check in the widgetMappers and return null from there if the check fails. Not sure if it's better than this though.